diff --git a/dist/cjs/server/staticMiddleware.js b/dist/cjs/server/staticMiddleware.js
index 18d16c1e211c070e9e88e896065906599da247dc..dfc026d299988ba5d0850db7586f2e4c17557278 100644
--- a/dist/cjs/server/staticMiddleware.js
+++ b/dist/cjs/server/staticMiddleware.js
@@ -69,7 +69,7 @@ const createStaticMiddleware = (options)=>{
         const pathname = c.req.path;
         if ('.js' !== external_path_default().extname(pathname)) return next();
         const prefixWithoutHost = removeHost(assetPrefix);
-        const prefixWithBundle = external_path_default().join(prefixWithoutHost, bundlesAssetPrefix);
+        const prefixWithBundle = external_path_namespaceObject.posix.join(prefixWithoutHost, bundlesAssetPrefix);
         if (!pathname.startsWith(prefixWithBundle)) return next();
         const pathnameWithoutPrefix = pathname.replace(prefixWithBundle, '');
         const filepath = external_path_default().join(pwd, bundlesAssetPrefix, pathnameWithoutPrefix);
@@ -77,7 +77,7 @@ const createStaticMiddleware = (options)=>{
         const fileResult = await external_fileCache_js_namespaceObject.fileCache.getFile(filepath);
         if (!fileResult) return next();
         c.header('Content-Type', "application/javascript");
-        c.header('Content-Length', String(fileResult.content.length));
+        c.header('Content-Length', String(Buffer.byteLength(fileResult.content)));
         return c.body(fileResult.content, 200);
     };
 };
diff --git a/dist/esm/server/staticMiddleware.mjs b/dist/esm/server/staticMiddleware.mjs
index 3a244d52116d28ffce31ef0e9929cd3eb6c1fec2..fa8833b39d39c68c2fc86e3d8cad324231e97cfe 100644
--- a/dist/esm/server/staticMiddleware.mjs
+++ b/dist/esm/server/staticMiddleware.mjs
@@ -63,7 +63,7 @@ var staticMiddleware_createStaticMiddleware = function(options) {
                             next()
                         ];
                         prefixWithoutHost = staticMiddleware_removeHost(assetPrefix);
-                        prefixWithBundle = path.join(prefixWithoutHost, bundlesAssetPrefix);
+                        prefixWithBundle = path.posix.join(prefixWithoutHost, bundlesAssetPrefix);
                         if (!pathname.startsWith(prefixWithBundle)) return [
                             2,
                             next()
@@ -90,7 +90,7 @@ var staticMiddleware_createStaticMiddleware = function(options) {
                             next()
                         ];
                         c.header('Content-Type', "application/javascript");
-                        c.header('Content-Length', String(fileResult.content.length));
+                        c.header('Content-Length', String(Buffer.byteLength(fileResult.content)));
                         return [
                             2,
                             c.body(fileResult.content, 200)
diff --git a/dist/esm-node/server/staticMiddleware.mjs b/dist/esm-node/server/staticMiddleware.mjs
index d3dd6972f242ef1e0af60a1441333200eace361c..b68c8f5e77fd63514955fccda3580ea9e1fc3901 100644
--- a/dist/esm-node/server/staticMiddleware.mjs
+++ b/dist/esm-node/server/staticMiddleware.mjs
@@ -26,7 +26,7 @@ const createStaticMiddleware = (options)=>{
         const pathname = c.req.path;
         if ('.js' !== path.extname(pathname)) return next();
         const prefixWithoutHost = removeHost(assetPrefix);
-        const prefixWithBundle = path.join(prefixWithoutHost, bundlesAssetPrefix);
+        const prefixWithBundle = path.posix.join(prefixWithoutHost, bundlesAssetPrefix);
         if (!pathname.startsWith(prefixWithBundle)) return next();
         const pathnameWithoutPrefix = pathname.replace(prefixWithBundle, '');
         const filepath = path.join(pwd, bundlesAssetPrefix, pathnameWithoutPrefix);
@@ -34,7 +34,7 @@ const createStaticMiddleware = (options)=>{
         const fileResult = await fileCache.getFile(filepath);
         if (!fileResult) return next();
         c.header('Content-Type', "application/javascript");
-        c.header('Content-Length', String(fileResult.content.length));
+        c.header('Content-Length', String(Buffer.byteLength(fileResult.content)));
         return c.body(fileResult.content, 200);
     };
 };
